.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,50 @@
+name: golangci-lint
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - main
+  pull_request:
+permissions:
+  contents: read
+  # Optional: allow read access to pull request. Use with `only-new-issues` option.
+  # pull-requests: read
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-go@v3
+        with:
+          go-version: 1.19
+      - uses: actions/checkout@v3
+      # Ginkgo and gomega are not required to pass linter.
+      - name: Install ginkgo
+        run: go install github.com/onsi/ginkgo/v2/ginkgo
+      - name: Install gomega
+        run: go get github.com/onsi/gomega/...
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v3
+        with:
+          # Optional: version of golangci-lint to use in form of v1.2 or v1.2.3 or `latest` to use the latest version
+          version: latest
+
+          # Optional: working directory, useful for monorepos
```

Comment:
just a suggestion, maybe worth removing all these optional comments to have a cleaner file and in the future in case we need one of them we can just add it.

Reply:
Good point. I removed lines we will not need and kept a couple which we may need to uncomment and use in the near future.